godot-ffi/src/lib.rs

```diff
@@ -447,6 +447,31 @@ pub fn is_main_thread() -> bool {
     }
 }
 
+/// Assign the current thread id to be the main thread.
+///
+/// This is required for platforms on which Godot runs the main loop on a different thread than the thread the library was loaded on.
+/// Android is one such platform.
+///
+/// # Safety
+///
+/// - must only be called after [`initialize`] has been called.
+pub unsafe fn make_main_thread() {
+    #[cfg(not(wasm_nothreads))]
+    {
+        if is_main_thread() {
+            // we don't have to do anything if the current thread is already the main thread.
+            return;
+        }
+
+        let thread_id = std::thread::current().id();
+
+        // SAFETY: initialize must have already been called before this function is called, so we have to clear the cell first.
+        unsafe { MAIN_THREAD_ID.clear() };
+        // SAFETY: we just called clear so the cell is no longer initialized.
+        unsafe { MAIN_THREAD_ID.set(thread_id) };
```

Comment:
Could probably be combined to 1 unsafe block? Explaining the set -> clear -> set cycle.